src/WingetCreateCLI/Commands/UpdateLocaleCommand.cs

```diff
@@ -0,0 +1,291 @@
+                        }
+                    }
+                    catch (ArgumentException)
+                    {
+                        Logger.ErrorLocalized(nameof(Resources.InvalidLocale_ErrorMessage));
+                        return false;
+                    }
+                }
+
+                Manifests updatedLocales = this.PromptAndUpdateExistingLocales(originalManifests);
+                Console.WriteLine();
+                EnsureManifestVersionConsistency(originalManifests);
+                DisplayUpdatedLocales(updatedLocales);
+
+                if (string.IsNullOrEmpty(this.OutputDir))
+                {
+                    this.OutputDir = Directory.GetCurrentDirectory();
+                }
+
+                string manifestDirectoryPath = SaveManifestDirToLocalPath(originalManifests, this.OutputDir);
+
+                if (ValidateManifest(manifestDirectoryPath))
+                {
+                    if (Prompt.Confirm(Resources.ConfirmGitHubSubmitManifest_Message))
+                    {
+                        return await this.LoadGitHubClient(true) ?
+                            (commandEvent.IsSuccessful = await this.GitHubSubmitManifests(
+                                originalManifests,
+                                this.GetPRTitle(originalManifests, null, nameof(UpdateLocaleCommand))))
+                            : false;
+                    }
+                    else
+                    {
+                        Logger.WarnLocalized(nameof(Resources.SkippingPullRequest_Message));
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+
+                return await Task.FromResult(commandEvent.IsSuccessful = true);
+            }
+            catch (Octokit.NotFoundException e)
+            {
+                Logger.ErrorLocalized(nameof(Resources.Error_Prefix), e.Message);
+                Logger.ErrorLocalized(nameof(Resources.OctokitNotFound_Error));
+                return false;
+            }
+            finally
+            {
+                TelemetryManager.Log.WriteEvent(commandEvent);
+            }
+        }
+
+        private static object PromptAllLocalesAsMenuSelection(Manifests originalManifests)
+        {
+            Dictionary<string, object> localeMap = new Dictionary<string, object>
+            {
+                { originalManifests.DefaultLocaleManifest.PackageLocale + " (" + Resources.DefaultLocale_MenuItem + ")", originalManifests.DefaultLocaleManifest },
+            };
+
+            foreach (var locale in originalManifests.LocaleManifests)
+            {
+                localeMap.Add(locale.PackageLocale, locale);
+            }
+
+            string selectedLocale = Prompt.Select(Resources.SelectExistingLocale_Message, localeMap.Keys.ToList());
+            return localeMap[selectedLocale];
+        }
+
+        private static void DisplayUpdatedLocales(Manifests manifest)
```

Comment:
It's not clear why a manifest would be considered 'updated' in this method. The only difference is that you are checking if its null. I'd rather just call the contents of this method in the code above rather than hiding it inside another private method.

Reply:
updatedLocales would contain all the locales the user has updated through the update-locale command. This method generates a preview of those "updated" locales. Since we don't know whether the updated locale would be a DefaultLocale or LocaleManifest before-hand, we null-check for its existence and print out that locale.

I can refactor this out to remove the method and just call the contents above, but since NewLocaleCommand also has a similar method `DisplayGeneratedLocales` , I would rather keep its counterpart for consistency since the overall flow is quite similar